### PR TITLE
Separate workshop page edit view

### LINKED ISSUE #67: Separate workshop page edit and view views
Currently if you have right to edit a workshop page you can't see how it looks like without logging out.

### PR DESCRIPTION
**Pull request by cytadela8**
This closes #67

### CODE REVIEW

File: wwwapp/urls.py [hunk 1]
```diff
@@ -28,9 +28,10 @@
     url(r'^article/(?P<name>[a-zA-Z0-9\-_]+)/$', views.article_view, name='article'),
     url(r'^articleNameList/$', views.article_name_list_view, name='articleNameList'),
     url(r'^addArticle/$', views.article_view, name='addArticle'),
-    url(r'^workshop/priv/(?P<name>[a-zA-Z0-9\-_]+)/$', views.workshop_view, name='workshop'),
     url(r'^workshop/(?P<name>[a-zA-Z0-9\-_]+)/$', views.workshop_page_view, name='workshop_page'),
-    url(r'^workshop/participants/(?P<name>[a-zA-Z0-9\-_]+)/$', views.workshop_participants_view, name='workshop_participants'),
+    url(r'^workshop/(?P<name>[a-zA-Z0-9\-_]+)/priv/$', views.workshop_view, name='workshop'),
+    url(r'^workshop/(?P<name>[a-zA-Z0-9\-_]+)/edit$', views.workshop_page_edit_view, name='workshop_page_edit'),
```
Comment (krzys-h) [inline wwwapp/urls.py]: Can we have any kind of consistency in these URLs please? All of them end with `/` except for these two...

Comment (cytadela8) [reply wwwapp/urls.py]: Consistency what is that? XD

Comment (krzys-h) [reply wwwapp/urls.py]: regularity, consistency, uniformity, conformity, symmetry, constancy, whichever one you prefer :P

**Comment by cytadela8** [comment]
#66 actually introduces permission for admins to edit workshops.

I will probably work on fixing the links tomorrow. This is not trivial. We should also use a navbar here.

**Comment by krzys-h** [comment]
Navbar below a navbar may look weird. I actually don't mind how it looks now as long as it works reasonably.

I know #66 introduces a permission for workshop pages and qualification, I just wasn't sure about workshop proposals.

The navbar links are actually correct already - the only ones that need changing are the ones on the workshop list.

**Comment by cytadela8** [comment]
Ok, so:

> If admin can edit proposals other than their own (and he should be able to, since that part contains info about the title, category and URL name, although I haven't checked what #66 does), that page should get it's read-only counterpart as well (and in that case, the link should default to viewing UNLESS it's your own workshop, to keep the current behavior which was really nice)

Let's do that as part of #66 

> The URL in "Your workshops" and "All workshops" should redirect directly to the editing page, not the preview page. The "Program" page has links to public pages if your goal is to browse those.

I will correct that in this PR. There is some non-trivial work needed to provide a correct link (there is one template for both own and all workshops)

File: templates/workshoplist.html [hunk 2]
```diff
@@ -31,7 +31,11 @@ <h2>{{ entry.year }}</h2>
                   </td>
                   <td>
                     {% if workshop.is_publicly_visible %}
-                      <a href="{% url 'workshop_page' workshop.name %}">{{ workshop.title }}</a>
```
Comment (krzys-h) [inline templates/workshoplist.html]: Wait, I'm confused now. It was supposed to be like this:
* Your workshops - editing page
* All workshops - editing page
* Program - public page

but you made it this:
* Your workshops - editing page
* All workshops - public page
* Program - public page

This makes no sense, the admin pages were supposed to default to editing, "Program" has links to public pages, why would you want to just view the public page from an admin panel

Comment (cytadela8) [reply templates/workshoplist.html]: This is actually correct.

Editing is part of #66. Currently even superadmins don't have the permission to edit anything in other users workshops. (unless they edit through Django admin panel) See implementation of `views.can_edit_workshop` function.
Under the implementation before this PR, admins would get the public page after clicking on link in all workshops.

Comment (krzys-h) [reply templates/workshoplist.html]: Ah, okay, makes sense - I forgot there is no editing part yet.

File: static/js/warsztatywww.js [hunk 3]
```diff
@@ -111,7 +111,7 @@ $(function () {
     $('[data-toggle="popover"]').popover();
 
     // Automatically hide 'Saved successfully' alerts after 4 seconds
-    $(".alert").delay(4000).fadeTo(500, 0).slideUp(500, function(){
+    $(".alert-info, .alert-success").delay(4000).fadeTo(500, 0).slideUp(500, function(){
```
Comment (grzegorz225) [inline static/js/warsztatywww.js]: Hard-coded pixel coordinates seems like a bad idea. Get the viewport size and divide by 2.

File: wwwapp/views.py [hunk 4]
```diff
@@ -205,78 +204,93 @@ def workshop_view(request, name=None):
...
-    context['form'] = form
     context['has_perm_to_edit'] = has_perm_to_edit
+    context['has_perm_to_view_details'] = has_perm_to_edit or has_perm_to_see_all
+
+    context['form'] = form
 
     return render(request, 'workshop.html', context)
 
 
+def can_edit_workshop(workshop, user):
```
Comment (grzegorz225) [inline wwwapp/views.py]: Typespec

File: wwwapp/views.py [hunk 5]
```diff
@@ -528,17 +542,17 @@ def article_name_list_view(request):
...
 @login_required()
 @permission_required('wwwapp.see_all_workshops', raise_exception=True)
 def all_workshops_view(request):
     workshops = Workshop.objects.all()
-    return render_workshops(request, 'Wszystkie warsztaty', workshops)
+    return render_workshops(request, 'Wszystkie warsztaty', False, workshops)
 
 
-def render_workshops(request, title, workshops):
+def render_workshops(request, title, link_to_edit, workshops):
```
Comment (grzegorz225) [inline wwwapp/views.py]: Why do we pass booleans to the link_to_edit parameter. Add typespecs.

File: static/js/warsztatywww.js (hunk 3 above)
Comment (krzys-h) [reply static/js/warsztatywww.js]: I don't see any pixels here, only time in ms :P

And stop complaining about things that were not changed in pull request reviews, that's not what they are for - open an issue or fix it yourself in another PR

File: wwwapp/views.py (hunk 5 above)
Comment (cytadela8) [reply wwwapp/views.py]: Becouse it's a boolean.

File: wwwapp/views.py (hunk 4 above)
Comment (cytadela8) [reply wwwapp/views.py]: There wasn't typespec here before.

**Pull request merged**